src/polyswarm/search.py

```diff
@@ -1,96 +1,74 @@
-@click.option('-b', '--without-bounties', is_flag=True, default=False,
-              help='Don\'t request bounties.')
-@click.argument('hashes', nargs=-1)
-@search.command('hash', short_help='search for hashes separated by space')
+@click.argument('hash_value', nargs=-1)
 @click.pass_context
-def hashes(ctx, hashes, hash_file, hash_type, without_metadata, without_bounties):
+def hashes(ctx, hash_value, hash_file, hash_type):
     """
     Search PolySwarm for files matching hashes
     """
-
     api = ctx.obj['api']
     output = ctx.obj['output']
 
-    hashes = parse_hashes(hashes, hash_type, hash_file)
-    try:
-        if hashes:
-            results = api.search(*hashes, with_instances=not without_bounties, with_metadata=not without_metadata)
-
-            # for json, this is effectively jsonlines
-            all_failed = True
-            for result in results:
-                output.search_result(result)
-                if not result.failed:
-                    all_failed = False
-
-            if all_failed:
-                sys.exit(1)
-        else:
-            raise click.BadParameter('Hash not valid, must be sha256|md5|sha1 in hexadecimal format')
-    except exceptions.UsageLimitsExceeded:
-        output.usage_exceeded()
-        sys.exit(1)
+    hashes_ = utils.parse_hashes(hash_value, hash_file=hash_file, hash_type=hash_type, log_errors=True)
+
+    results = api.search(*hashes_)
+    utils.validate_results(results)
 
+    # for json, this is effectively jsonlines
+    for result in results:
+        output.artifact_instance(result)
 
+
+@search.command('metadata', short_help='search metadata of files')
 @click.option('-r', '--query-file', help='Properly formatted JSON search file', type=click.File('r'))
-@click.option('-m', '--without-metadata', is_flag=True, default=False,
-              help='Don\'t request artifact metadata.')
-@click.option('-b', '--without-bounties', is_flag=True, default=False,
-              help='Don\'t request bounties.')
 @click.argument('query_string', nargs=-1)
-@search.command('metadata', short_help='search metadata of files')
 @click.pass_context
-def metadata(ctx, query_string, query_file, without_metadata, without_bounties):
+def metadata(ctx, query_string, query_file):
 
     api = ctx.obj['api']
     output = ctx.obj['output']
 
     try:
         if len(query_string) >= 1:
-            queries = [MetadataQuery(q, False, api) for q in query_string]
+            queries = [resources.MetadataQuery(q, False, api) for q in query_string]
         elif query_file:
-            # TODO support multiple queries in a file?
-            queries = [MetadataQuery(json.load(query_file), True, api)]
+            # TODO: support multiple queries in a file?
+            queries = [resources.MetadataQuery(json.load(query_file), True, api)]
         else:
             logger.error('No query specified')
-            return 0
-    except json.decoder.JSONDecodeError:
+            return 2
```

Comment:
If we intend to use this return values as exit codes, we should either make them generic (e.g `1`) or return an entry from an enum of special values because `2` is often used by the shell itself for parse errors.